contracts/RewardController.sol

```diff
@@ -0,0 +1,276 @@
+
+        if (_sharesChange != 0) {
+            if (_isDeposit) {
+                userShares += _sharesChange;
+            } else {
+                userShares -= _sharesChange;
+            }
+        }
+        rewardDebt[_vault][_user] = userShares * rewardPerShare / REWARD_PRECISION;
+    }
+
+    /**
+    * @notice Called by the vault to update a user claimable reward after deposit or withdraw.
+    * This call should never revert
+    * @param _user The user address to updare rewards for
+    * @param _sharesChange The user of shared the user deposited or withdrew
+    * @param _isDeposit Whether user deposited or withdrew
+    */
+    function commitUserBalance(address _user, uint256 _sharesChange, bool _isDeposit) external {
+        _commitUserBalance(msg.sender, _user, _sharesChange, _isDeposit);
+    }
+
+    /**
+    * @notice Transfer to the specified user their pending share of rewards.
+    * @param _vault The vault address
+    * @param _user The user address to claim for
+    */
+    function claimReward(address _vault, address _user) external {
+        _commitUserBalance(_vault, _user, 0, true);
+
+        uint256 userUnclaimedReward = unclaimedReward[_vault][_user];
+        if (userUnclaimedReward > 0) {
+            unclaimedReward[_vault][_user] = 0;
+            rewardToken.safeTransfer(_user, userUnclaimedReward);
+        }
+
+        emit ClaimReward(_vault, _user, userUnclaimedReward);
+    }
+
+    /**
+    * @notice Calculate rewards for a vault by iterating over the history of totalAllocPoints updates,
+    * and sum up all rewards periods from vault.lastRewardBlock until current block number.
+    * @param _vault The vault address
+    * @param _fromBlock The block from which to start calculation
+    * @return reward
+    */
+    function getVaultReward(address _vault, uint256 _fromBlock) public view returns(uint256 reward) {
+        if (globalVaultsUpdates.length == 0) {
+            return 0;
+        }
+        uint256 vaultAllocPoint = vaultInfo[_vault].allocPoint;
+        uint256 i = vaultInfo[_vault].lastProcessedVaultUpdate;
+        for (; i < globalVaultsUpdates.length-1; i++) {
+            uint256 nextUpdateBlock = globalVaultsUpdates[i+1].blockNumber;
+            reward =
+            reward + getRewardForBlocksRange(_fromBlock,
+                                            nextUpdateBlock,
+                                            vaultAllocPoint,
+                                            globalVaultsUpdates[i].totalAllocPoint);
+            _fromBlock = nextUpdateBlock;
+        }
+        return reward + getRewardForBlocksRange(_fromBlock,
+                                                block.number,
+                                                vaultAllocPoint,
+                                                globalVaultsUpdates[i].totalAllocPoint);
+    }
+
+    function getRewardForBlocksRange(uint256 _fromBlock, uint256 _toBlock, uint256 _allocPoint, uint256 _totalAllocPoint)
```

Comment:
So making it public almost costs nothing, so I guess there is no compelling reason to make it internal.

WE are also _testing_ the function in the tests, so that is actually good :)

Reply:
ok but i didn't include this and `getGlobalVaultsUpdatesLength` in the interface, makes sence?